### PR TITLE
[Word] (1.5) Updates

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**

### CODE REVIEW

File: samples/word/50-document/manage-styles.yaml [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+            if (newStyleName == "") {
+              console.warn("Enter a style name to add.");
+              return;
+            }
+
+            const style = context.document.getStyles().getByNameOrNullObject(newStyleName);
+            style.load();
+            await context.sync();
+
+            if (style.nameLocal) {
```
Comment (jipyua) [inline samples/word/50-document/manage-styles.yaml:44]: I think a more clear way is to check something like style.isNullObject()

File: samples/word/50-document/manage-styles.yaml [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            if (styleName == "") {
+              console.warn("Enter a style name to get properties.");
+              return;
+            }
+
+            const style = context.document.getStyles().getByNameOrNullObject(styleName);
+            style.load();
+            await context.sync();
+
+            if (!style.nameLocal) {
```
Comment (jipyua) [inline samples/word/50-document/manage-styles.yaml:70]: style.isNullObject()

File: samples/word/50-document/manage-styles.yaml [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+            if (styleName == "") {
+              console.warn("Enter a style name to apply.");
+              return;
+            }
+
+            const style = context.document.getStyles().getByNameOrNullObject(styleName);
+            style.load();
+            await context.sync();
+
+            if (!style.nameLocal) {
```
Comment (jipyua) [inline samples/word/50-document/manage-styles.yaml:96]: similar to use isNullObject()

File: samples/word/50-document/manage-styles.yaml [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+                "Start"
+              );
+              const paragraph = body.paragraphs.getFirst();
+              paragraph.style = style.nameLocal;
+              console.log(`'${styleName}' style applied to first paragraph.`);
+            }
+          });
+        }
+
+        async function setFontColor() {
```
Comment (jipyua) [inline samples/word/50-document/manage-styles.yaml:114]: the cases will grow more and more, I think this SetFontColor and the following SetFontSize can be merged as one for simplity

File: samples/word/50-document/manage-styles.yaml [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+            if (!style.nameLocal) {
+              console.warn(`There's no existing style with the name '${styleName}'.`);
+            } else {
+              style.font.size = 20;
+              console.log(`Successfully updated font size of the '${styleName}' style.`);
+            }
+          });
+        }
+
+        async function setParagraphFormatLeftIndent() {
```
Comment (jipyua) [inline samples/word/50-document/manage-styles.yaml:159]: similiarly, I think the two paragraphFormat cases can be merged into one.

File: samples/word/50-document/save-close.yaml [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    content: |
+        $("#save-no-prompt").click(() => tryCatch(saveNoPrompt));
+        $("#save-after-prompt").click(() => tryCatch(saveAfterPrompt));
+        $("#close-after-save").click(() => tryCatch(closeAfterSave));
+        $("#close-no-save").click(() => tryCatch(closeWithoutSave));
+        $("#save").click(() => tryCatch(save));
+        $("#close").click(() => tryCatch(close));
+
+        async function saveNoPrompt() {
+          // Save the document with the file name provided.
```
Comment (jipyua) [inline samples/word/50-document/save-close.yaml:18]: better to add a comment: the file name only take effect when the file hasn't be saved before.

File: samples/word/99-preview-apis/manage-fields.yaml [hunk 7]
```diff
@@ -0,0 +1,347 @@
+id: word-manage-fields
```
Comment (jipyua) [inline samples/word/99-preview-apis/manage-fields.yaml:1]: if this preview script is just for Fileld.showCodes then I think there is no need to keep this one for several reasons:
1. showCodes is not that important, we do not necessarily add code samples for each of the APIs we are providing.
2. there are big limitations on wac to implement this, so I am afraid so far there is no plan to further ship it in the future, so we don't want to promote it.

File: samples/word/10-content-controls/content-control-onadded-event.yaml [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+            console.log("Content controls inserted: " + paragraphs.items.length);
+
+            await context.sync();
+          });
+        }
+
+        async function deregisterEventHandler() {
+          await Word.run(async (context) => {
+            await Word.run(eventContext.context, async (context) => {
```
[the remark this answers is not in the document]
Comment (xiaoyuMS) [reply samples/word/10-content-controls/content-control-onadded-event.yaml:55]: No need.
<pre>
async function deregisterEventHandlers() {
    await Word.run(eventContext.context, async (context) => {
      eventContext.remove();
      await context.sync();
      eventContext = null;
    });
}
</pre>

File: samples/word/10-content-controls/content-control-onadded-event.yaml [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+            await context.sync();
+
+            console.log("Added event handler for when content controls are added.");
+          });
+        }
+
+        async function contentControlAdded(event: Word.ContentControlAddedEventArgs) {
+          await Word.run(async (context) => {
+            // Display the added content control's ID.
+            console.log(`${event.eventType} event detected. ID of content control that was added: ${event.ids[0]}`);
```
[the remark this answers is not in the document]
Comment (xiaoyuMS) [reply samples/word/10-content-controls/content-control-onadded-event.yaml:30]: Could we comment it as "Display the first added content control's ID"? There can be more than one ID in the event arguments.

File: samples/word/50-document/manage-fields.yaml [hunk 10]
```diff
@@ -1,30 +1,76 @@
...
+            var range = context.document.getSelection().getRange();
+
+            const field = range.insertField(Word.InsertLocation.before, Word.FieldType.date, '\\@ "M/d/yyyy h:mm am/pm"', true);
+
+            field.load("result,code");
+            await context.sync();
+
+            if (field.isNullObject) {
+              console.log("There are no fields in this document.");
+            } else {
```
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply samples/word/50-document/manage-fields.yaml:33]: Opting to keep as it provides a better user experience

File: samples/word/50-document/manage-fields.yaml [hunk 11]
```diff
@@ -1,30 +1,76 @@
...
+          // Inserts an Addin field before selection.
+          await Word.run(async (context) => {
+            var range = context.document.getSelection().getRange();
+
+            const field = range.insertField(Word.InsertLocation.before, Word.FieldType.addin);
+            field.load("result,code");
+            await context.sync();
+
+            if (field.isNullObject) {
+              console.log("There are no fields in this document.");
```
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply samples/word/50-document/manage-fields.yaml:50]: same here - better user experience

File: samples/word/99-preview-apis/manage-fields.yaml (hunk 7 above)
Comment (ElizabethSamuel-MSFT) [reply samples/word/99-preview-apis/manage-fields.yaml]: removed

File: samples/word/10-content-controls/content-control-onadded-event.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (jingxuan0109) [reply samples/word/10-content-controls/content-control-onadded-event.yaml]: We could change it.
```   
for(let index=0; index<event.ids.length; index++){
      console.log(`${event.eventType} event detected. ID of content control that was added: ${event.ids[index]}`);
    }
```

**Review by jipyua — APPROVED** [review]

**Review by jipyua — APPROVED** [review]

**Pull request merged**